src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -10,6 +10,18 @@ namespace System.IO
 {
     public static class FileSystemAclExtensions
     {
+        /// <summary>Creates a new directory, ensuring it is created with the specified directory security.</summary>
+        /// <param name="directoryInfo">The object describing a directory that does not exist in disk yet.</param>
+        /// <param name="directorySecurity">An object that determines the access control and audit security for the directory.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="directorySecurity" /> is <see langword="null" />.</exception>
+        public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity directorySecurity)
+        {
+            if (directorySecurity == null)
```

Comment:
@JeremyKuhne should we make this `if (string.IsNullOrWhiteSpace(directorySecurity))` instead? Or do we let `FileSystem.CreateDirectory` handle the empty/whitespace case?

Reply:
Not sure what you mean. `directorySecurity` isn't a string.

Reply:
Oops my bad. I wrote this before my morning coffee.

Reply:
Actually, I remember why I wrote this _apparently dumb_ question: It was directed towards the `directoryInfo.FullName` argument, which we pass to the CreateDirectory method, but we don't check if it's NullOrWhiteSpace, we just check if the directory exists.

Reply:
Yeah, I think we're good leaving that to the DirectoryInfo construction